subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -563,6 +565,15 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 				&params->bssid[2], &params->bssid[3],
 				&params->bssid[4], &params->bssid[5]);
 			break;
+		case 't':
+			if (iface_mode == WIFI_MODE_INFRA) {
+				params->timeout = strtol(optarg, &endptr, 10);
+				if (*endptr != '\0') {
+					PR_ERROR("Invalid timeout: %s\n", optarg);
+					return -EINVAL;
+				}
+			}
```

Comment:
Should there be an error message if mode is not `INFRA` ?

Reply:
We can add a warning log. Since the upstream PR has already been merged, will take this up in a separate PR.

Reply:
Please file a Jira for the same.